clientd/src/lib.rs

```diff
@@ -12,18 +14,55 @@ use tokio::sync::{broadcast, mpsc, oneshot};
 mod rpc;
 pub mod server;
 
-// TODO: remove pub and use this type to enforce invariants
-#[derive(Debug, Serialize)]
-pub struct CallbackResponse(pub Result<String, RpcError>);
+#[derive(Clone)]
+pub struct EventSubscribers(HashMap<EventKey, broadcast::Sender<Event>>);
```

Comment:
One sentence docstring comment please! It take a 20 seconds, and help a lot. :)

Reply:
yeah I know xd will do doc comments next PR and then try to do it immediately

Reply:
BTW. Is there any important advantage of doing this, over using a single `broadcast::Sender<EventKey, Event>` and letting subscribers filter what they need?

Reply:
I was thinking about this approach too ( this would also mean we could have one `/subscribe` endpoint where the api consumer could give a list of events in the params) but this is not scalable because noisy events would clog other subscribers channel because they all share one. ok while writing this I notice the dispatcher is the same bottle neck anyway but I still think going for a design like this makes more sense, it just feels cleaner)